extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/groupby/GroupByQueryKit.java

```diff
@@ -166,34 +170,83 @@ public QueryDefinition makeQueryDefinition(
         partitionBoost
     );
 
-    queryDefBuilder.add(
-        StageDefinition.builder(firstStageNumber + 1)
-                       .inputs(new StageInputSpec(firstStageNumber))
-                       .signature(resultSignature)
-                       .maxWorkerCount(maxWorkerCount)
-                       .shuffleSpec(
-                           shuffleSpecFactoryPostAggregation != null
-                           ? shuffleSpecFactoryPostAggregation.build(resultClusterBy, false)
-                           : null
-                       )
-                       .processorFactory(new GroupByPostShuffleFrameProcessorFactory(queryToRun))
-    );
+    // the result signature might change
+    // if window shufle spec is added
+    // say the output signature was d0, d1
+    // But shuffle spec for window was d1
+    // create the shufflespec from the column in the context
+    // and sort after wards to ensure prefix of shuffle is in row signature
+    final ShuffleSpec nextShuffleWindowSpec;
```

Comment:
I feel this complexity should be added in 
```
 ClusterBy resultClusterBy = computeResultClusterBy(
        queryToRun,
        segmentGranularity,
        partitionBoost
    );
    RowSignature resultSignature = computeResultSignature(
        queryToRun,
        segmentGranularity,
        resultClusterBy,
        partitionBoost
    );
    ```

Reply:
I did not understand what you mean here. This is already added in and in L190-195 I am creating a new shuffle spec with the columns in the resultClusterBy and the columns in the partition by of the over

Reply:
I mean the method `computeResultClusterBy` should have the windowing logic pushed in also. 
Currently it gives you something and then you mutate it based on the windowing requirement.